recipe/meta.yaml

```diff
@@ -24,25 +25,57 @@ requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-    - patch  # [not win]
   host:
     - python
     - pip
     - cython >=0.29.32,<3
-    - numpy   1.19  # [not (osx and arm64) and (py==38 or py==39)]
-    - numpy   1.21  # [(osx and arm64) and (py==38 or py==39)]
-    - numpy   1.21  # [py==310]
+    - numpy {{ numpy }}
     - setuptools >=51.0.0
     - wheel
   run:
     - python
-    - {{ pin_compatible('numpy') }}
+    # Dependencies
+    # see https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#dependencies
+    - numpy >={{ numpy }},<2.0a0
     - python-dateutil >=2.8.1
     - pytz >=2020.1
     # Recommended dependencies to achieve large speedups
-    # see https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#recommended-dependencies
-    - numexpr >=2.7.1
-    - bottleneck >=1.3.1
+    # see https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#recommended-dependencies
+    - numexpr >=2.7.3
+    - bottleneck >=1.3.2
+  run_constrained:
+    # Optional dependencies
+    # see https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#optional-dependencies
+    - tzdata >=2022a
+    - matplotlib >=3.3.2
+    - jinja2 >=3.0.0
+    - tabulate >=0.8.9
+    - scipy >=1.7.1
+    - numba >=0.53.1
+    - xarray >=0.19.0
+    - xlrd >=2.0.1
+    - xlwt >=1.3.0
+    - xlsxwriter >=1.4.3
+    - openpyxl >=3.0.7
+    - pyxlsb >=1.0.8
+    - beautifulsoup4 >=4.9.3
+    - html5lib >=1.1
+    - lxml >=4.6.3
+    - sqlalchemy >=1.4.16
+    - psycopg2 >=2.8.6
+    - pymysql >=1.0.2
+    - pytables >=3.6.1
+    - blosc >=1.21.0
+    - fastparquet >=0.4.0
```

Comment:
missing `zlib` https://pandas.pydata.org/pandas-docs/version/1.5.1/getting_started/install.html#other-data-sources ?

Reply:
I did not list it because there was no version constraint on zlib.